components/calendar/src/types.rs

```diff
@@ -448,7 +448,13 @@ impl Month {
     /// The input saturates at 99.
     pub const fn new(number: u8) -> Self {
         Self {
-            number: if number > 99 { 99 } else { number },
+            number: if number > 99 {
+                99
+            } else if number == 0 {
+                1
+            } else {
+                number
+            },
```

Comment:
Issue: We should allow M00 and M00L

Reply:
`MonthCode` doesn't allow it, and never did: https://docs.rs/icu_calendar/2.0.0/src/icu_calendar/types.rs.html#154

Reply:
In 2.0 they all returned the same error, UnknownMonthCode, which is the same in 2.1.

There probably won't ever be a M00, but there could potentially be a M00L, i.e. a year that starts with a leap month. It doesn't happen in Chinese or Hebrew but it could occur in Hindu, I think.

Reply:
and if that hypothetical situation ever happens in ICU4X, we can start allowing `M00L`.

From fields returns `MonthNotInCalendar` for `M00`, I think that's wrong. because a month 0 for 1-indexed months is not just not in calendar, it fundamentally doesn't exist.

Reply:
It sounds like if we reject M00 and accept M00L then we're both satisfied.

Reply:
If you don't want to design for hypotheticals, then Month should only accept month codes that are valid in at least one CLDR calendar, and drop M14..M99 and M13L. My preference is to allow those codes plus M00L.

Reply:
happy to do that